Separate wire protocol from internal models

Arkatufus opened the pull request:
## Changes

Separate Akka.Cluster.Metrics wire format model from internal model.
We were using the proto generated classes before. All protobuf generated codes are now internal.

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Changes in public API reviewed, if any.

Aaronontheweb commented on src/contrib/cluster/Akka.Cluster.Metrics/Serialization/Proto/ClusterMetricsMessages.proto:
```diff
@@ -14,50 +14,50 @@ option optimize_for = SPEED;
  * Metrics Gossip Envelope
  */
-message MetricsGossipEnvelope {
+message MetricsGossipEnvelopeProto {
```
can't rename messages - strongly prefer to break the internal C# types.

Aaronontheweb replied:
Move these into .`Proto` namespace

Aaronontheweb replied:
TLDLR the above breaks wire compat

Arkatufus commented:
OK, fixxed. Moved the .proto file to the general protobuf folder, modified the build script, and moved the proto classes to the standardized `.Proto.Msg` namespace

Aaronontheweb commented:
This definitely passes wire compat now - wire types are unchanged.
